app/controllers/concerns/saml_idp_auth_concern.rb

```diff
@@ -187,6 +187,7 @@ def current_issuer
 
   def request_url
     url = URI.parse request.original_url
+    url.path = url.path.gsub('post', '') if url.path.match?(/authpost/)
```

Comment:
Not sure how to test this, exactly.

Reply:
For testing: maybe break this into a separate method so it can be called directly in a spec, ex `remap_auth_post_path(url)`

And also I think that we can tighten up some of the assumptions, and maybe if we construct the URL more directly it might be clearer? 

```suggestion
    path_match = url.path.match(%r{/api/saml/authpost(?<year>\d+)})
    if path_match
      url.path = "/api/saml/auth#{path_match[:year]}"
    end
```

Reply:
Good suggestion, done! I also figured out how to test this - I'll comment separately in the diff.